key.util/src/main/java/org/key_project/util/EqualsModProofIrrelevancyUtil.java

```diff
@@ -99,7 +99,10 @@ public static int hashCodeIterable(Iterable<? extends EqualsModProofIrrelevancy>
     public static boolean compareImmutableLists(
             ImmutableList<? extends EqualsModProofIrrelevancy> a,
             ImmutableList<? extends EqualsModProofIrrelevancy> b) {
-        if (a.size() != b.size()) {
+        if (a == b || (a == null && b.size() == 0) || (b == null && a.size() == 0)) {
```

Comment:
Doesn't this throw a NPE if b is null (as well as if a == null and b.size() > 0)? Also, it seems strange that empty lists and null lists should be equal ...

Reply:
If a == null and b == null the first check `a == b` succeeds. Therefore at least one of a, b will be non-null when checking the rest of the condition.

If a is null and b.size() > 0, the next check in the function returns false.

I'm not sure this change is required at this point, I will check again.

Reply:
Ah, you are right, I was confused here ;-) We can leave it as is.